### PR TITLE
Added sentence window retrieval

### PR DESCRIPTION
**Pull request by deltaaruna**
Added sentence window retrieval support 
Based on the llamaindex Docs - https://docs.llamaindex.ai/en/stable/examples/node_postprocessor/MetadataReplacementDemo/

Add enviroment variable USE_SENTENCE_WINDOW_RETRIEVAL to enable.
Add enviroment variable WINDOW_SIZE to specify the window size.

@nilaan-effectz or @iujinasena pls review.

### CODE REVIEW

File: effectzgpt/backend/app/engine/generate.py [hunk 1]
```diff
@@ -31,14 +32,20 @@ def get_doc_store():
...
-            ),
-            Settings.embed_model,
-        ],
+            ))
+    transformations.append(Settings.embed_model)
+
+    if os.getenv("USE_SENTENCE_WINDOW_RETRIEVAL", "True").lower() == "true":
+        transformations.append(get_sentence_window_node_parser())
+    
+    transformations.append(get_sentence_window_node_parser())
```
Comment (iujinasena) [inline effectzgpt/backend/app/engine/generate.py]: The same transformation is applied within the IF statement and outside the IF statement.

**Review by iujinasena — CHANGES_REQUESTED** [review]

File: effectzgpt/backend/app/engine/generate.py (hunk 1 above)
Comment (deltaaruna) [reply effectzgpt/backend/app/engine/generate.py]: Thank you for noticing. Will remove the second statement.

Comment (deltaaruna) [reply effectzgpt/backend/app/engine/generate.py]: Requested change have been implemented @iujinasena pls review.

File: effectzgpt/backend/app/engine/node_postprocessors.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from llama_index.core.postprocessor import MetadataReplacementPostProcessor
+from app.engine.constants import DEFAULT_META_DATA
```
Comment (iujinasena) [inline effectzgpt/backend/app/engine/node_postprocessors.py]: DEFAULT_META_DATA is not defined in constants.py

**Review by iujinasena — CHANGES_REQUESTED** [review]

File: effectzgpt/backend/app/engine/node_postprocessors.py (hunk 2 above)
Comment (deltaaruna) [reply effectzgpt/backend/app/engine/node_postprocessors.py]: @iujinasena this has been fixed.

**Review by iujinasena — APPROVED** [review]

**Pull request merged**